packages/frontapp/src/app/components/CarouselWrapper/index.tsx

```diff
@@ -0,0 +1,75 @@
+import styled from '@emotion/styled';
+import { dimensions } from '@mimir/ui-kit';
+import { FC, ReactNode } from 'react';
+import { Swiper } from 'swiper/react';
+import ControlPanel from './ControlPanel';
+import SliderButtons from './SliderButtons';
+import { useMediaQuery } from 'react-responsive';
+
+import 'swiper/css';
+
+const DESKTOP_SLIDES_PER_VIEW = 5;
+const LAPTOP_SLIDES_PER_VIEW = 4;
```

Comment:
Maybe there is a way to make this number calculated based on known card width, distance between cards and container width?

Reply:
This is pretty rare condition when the user will check the library with its desktop at all, but no, there are no any other options rather that using swipers `breakpoints`. I could add listener to the window resize, but it is not cost it and could lead later to memory leaks. What can I do here is to add one more breakpoint, but it will make buttons logic too complicated

Reply:
As you can see, there are pretty many other issues with `4k` as we didn't discussed how it will look like on such resolution, many different pages and containers has `100%` width which looks pretty ugly on `4k`

Reply:
Full HD is also not looking too great.
Try this
`  const containerRef = useRef<HTMLDivElement>(null);
  const [slidesPerView, setSliderCount] = useState(5);
  const SLIDE_WIDTH = 224;
  const SLIDE_GAP = 32;
  useEffect(() => {
    if (containerRef.current) {
      setSliderCount(Math.ceil((containerRef.current.clientWidth + SLIDE_GAP) / SLIDE_WIDTH));
    }
  }, [containerRef.current]);`
And then
`<CustomSwiper spaceBetween={SLIDE_GAP} slidesPerView={slidesPerView} >`

Reply:
Forgot one thing - need to wrap in div:
```
<div ref={containerRef}>
<CustomSwiper spaceBetween={SLIDE_GAP} slidesPerView={slidesPerView} >
</div>
```

Reply:
I've already got it) Thanks